api.ts

```diff
@@ -2516,7 +2525,11 @@ export function maybeRewriteConnectionForFormula<
 function listPropertiesWithAutocompleteFunctions(schema: ObjectSchemaDefinition<string, string>): string[] {
   const result: string[] = [];
   for (const propertyName of Object.keys(schema.properties)) {
-    const {autocomplete} = schema.properties[propertyName];
+    const propertySchema = maybeUnwrapArraySchema(schema.properties[propertyName]);
+    if (propertySchema?.codaType !== ValueHintType.SelectList) {
+      continue;
+    }
```

Comment:
would it be better to throw if autocomplete is defined but the coda type isn't select list? or maybe you already did this somewhere else. I'll keep reading.

Reply:
There's a zod error in that scenario, but it was missing a unit test which I just added. The zod validation error will say, `You must set "codaType" to ValueHintType.SelectList when setting an "autocomplete" property`